horreum-backend/src/main/java/io/hyperfoil/tools/horreum/changedetection/HunterEDivisiveModel.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    @Override
+    public void analyze(List<DataPointDAO> dataPoints, JsonNode configuration, Consumer<ChangeDAO> changeConsumer) {
+
+        TmpFiles tmpFiles = null;
+
+        try {
+            try {
+                tmpFiles = TmpFiles.instance();
+            } catch (IOException e) {
+                log.error("Could not create temporary file for Hunter eDivisive algorithm", e);
+                return;
+            }
+
+            assert tmpFiles.inputFile != null;
+
+            try (final FileWriter fw = new FileWriter(tmpFiles.inputFile, true);
+                 final PrintWriter pw = new PrintWriter(fw);) {
+
+                Collections.reverse(dataPoints);
+
+                //write out csv fields
+                pw.println(Arrays.stream(HEADERS).collect(Collectors.joining(",")));
+                dataPoints.forEach(dataPointDAO -> pw.println(String.format("%.2f,%s,%d", dataPointDAO.value, dataPointDAO.timestamp.toString(), dataPointDAO.id)));
+
+            } catch (IOException e) {
+                log.error("Could not create file writer for Hunter eDivisive algorithm", e);
+                return;
+            }
+
+            log.debugf("created csv output : %s", tmpFiles.inputFile.getAbsolutePath());
+
+            if (!validateInputCsv(tmpFiles)) {
+                log.errorf("could not validate: %s", tmpFiles.inputFile);
+                return;
+            }
+
+            DataPointDAO firstDatapoint = dataPoints.get(0);
+
+            processChangePoints(
+                    (dataPointID) -> dataPoints.stream().filter(dataPoint -> dataPoint.id.equals(dataPointID)).findFirst(),
+                    changeConsumer,
+                    tmpFiles,
+                    firstDatapoint.timestamp
+            );
+        } finally {
+            if (tmpFiles != null) {
+                cleanupTmpFiles(tmpFiles);
+            }
+        }
+    }
+
+    protected void cleanupTmpFiles(TmpFiles tmpFiles) {
+        if( tmpFiles.tmpdir.exists() ) {
+            clearDir(tmpFiles.tmpdir);
+        } else {
+            log.debugf("Trying to cleanup temp files, but they do not exist!");
+        }
+    }
+
+
+    private void clearDir(File dir){
+        Arrays.stream(dir.listFiles()).forEach(file -> {
+            if ( file.isDirectory() ){
+                clearDir(file);
+            }
+            file.delete();
+        });
+        if(!dir.delete()){
+            log.errorf("Failed to cleanup up temporary files: %s", dir.getAbsolutePath());
+        }
+    }
+
+    protected void processChangePoints(Function<Integer, Optional<DataPointDAO>> changePointSupplier, Consumer<ChangeDAO> changeConsumer, TmpFiles tmpFiles, Instant sinceInstance) {
+        String command = "hunter analyze horreum --since '" + sinceInstance.toString() + "'";
+        log.debugf("Running command: %s", command);
+
+
+        List<String> results = executeProcess(tmpFiles, false, "bash", "-l", "-c", command);
+
+        if (results.size() > 3) {
```

Comment:
I assume this if statement comes from how Hunter generates results, right? Maybe it is worth to add a comment or reference where this is stated, wdyt?

Reply:
Have added comments about the output file structure

Reply:
Sounds great, thanks!